doc/coq.rst

```diff
@@ -105,12 +105,20 @@ The semantics of the fields are:
 - Your Coq theory can depend on other theories by specifying them in the
   ``<coq_theories>`` field. Dune then passes to Coq the corresponding flags for
   everything to compile correctly (this corresponds to the ``-Q`` flag for Coq).
-  
-  As of today, we only support composition with libraries defined in the same
-  scope (i.e., under the same :ref:`dune-project` domain). This restriction will
-  be lifted in the future. Note that composition with the Coq standard library
-  is supported, but in this case the ``Coq`` prefix has been made available in a
-  qualified way, since :ref:`Coq lang 0.2<coq-lang>`.
+
+  You may also depend on theories belonging to another :ref:`dune-project` as long
+  as they share a common scope under another :ref:`dune-project` file or a
+  :ref:`dune-workspace` file.
+
+  Doing so can be a simple as placing a Coq project within the scope of another.
+  This process is termed *composition*. See the :ref:`inter-project
+  composition<example-inter-project-theory>` example. Inter-project composition
+  has been available since :ref:`Coq lang 0.4<coq-lang>`.
+
+  As of today, Dune cannot depend on installed Coq theories. This restriction
+  will be lifted in the future. Note that composition with the Coq standard
```

Comment:
Maybe add "In practice, this means that a dependency `(theory mod.lib ...)` won't be able to be resolved when that theory is installed in `$lib/coq/user-contrib` or in some path in `COQPATH`. This makes `dune build -p pkg` fail for Coq packages."

Reply:
Added some clarification.